mk/compile.mk

```diff
@@ -50,9 +50,12 @@ WARNS		?= 3
 
 comp-cflags$(sm)	+= $(comp-cflags-warns-$(WARNS))
 
+CHECK ?= sparse
```

Comment:
The "check" is only for sparse and it will be like that in the future also I guess? How about just replace the word "check" here and below with "sparse" (a variable). Doing so you make it more clear what a particular section in the makefile is about.